test/test_sparse.py

```diff
@@ -2177,6 +2177,36 @@ def test_factory_empty_indices(self):
         expected_indices = torch.empty((4, 0), dtype=torch.long, device=device)
         self.assertEqual(tensor._indices(), expected_indices)
 
+    def test_sparse_eye(self):
+        device = 'cuda' if self.is_cuda else 'cpu'
+
+        tensor = torch.eye(3, 4, layout=torch.sparse_coo, device=device)
```

Comment:
Can you please call these tests in a loop for different datatypes? If you use (3,4), (4,3), (3,3) sizes your expects can remain the same throughout. 
Also, please test the `out` variant of the function (`torch.eye(3,3, layout=..., out=...)`). 
Also, please check the size of the result.

Reply:
Will update the test case as per the suggestion.

Reply:
Oops, sorry, I committed my suggestion accidentally, but  I did not actually check that it runs, feel free to revert